zap/src/main/java/org/zaproxy/zap/core/scanner/InputVector.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.core.scanner;
+
+/**
+ * {@code InputVector} class wraps the parameters which are used to modify the {@code HttpMessage}.
+ * It is specifically used for updating multiple parameters of {@code HttpMessage}
+ *
+ * @author preetkaran20@gmail.com KSASAN
+ */
+public class InputVector {
+    private int position;
+    private String name;
+    private String value;
+    private PayloadFormat payloadFormat;
+
+    InputVector(int position, String name, String value, PayloadFormat payloadFormat) {
+        this.position = position;
+        this.name = name;
+        this.value = value;
+        this.payloadFormat = payloadFormat;
+    }
+
+    public int getPosition() {
+        return position;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public PayloadFormat getPayloadFormat() {
+        return payloadFormat;
+    }
+
+    /**
+     * {@code PayloadFormat} represents the type of InputVectors's name and value. This is useful
+     * for the use-cases like say there is a parameter which is a URL then caller can itself
+     * encode/escape it so that it can be directly used to modify {@code HttpMessage} or {@code
+     * Variant} has to escape it.
+     */
+    public static enum PayloadFormat {
```

Comment:
- Remove static.
 - Normally this would be declared above, before the instance variables.